apps/events/migrations/0014_move_schedule_messages_off_periodic_tasks.py

```diff
@@ -0,0 +1,120 @@
+    if periodic_task.one_off:
+        schedule = periodic_task.clocked
+        frequency = 1
+        time_period = None
+        repetitions = 1
+        next_trigger_date = schedule.clocked_time
+        total_triggers = 0
+    else:
+        schedule = periodic_task.interval
+        frequency = schedule.every
+        time_period = schedule.period
+        repetitions = calculate_remaining_repetitions(periodic_task, current_time) if not task_complete else 0,
+        next_trigger_date = calculate_next_run(periodic_task, current_time)  if not task_complete else None,
+        total_triggers = calculate_total_triggers(periodic_task, current_time)
+
+    return frequency, time_period, repetitions, total_triggers, next_trigger_date
+
+
+class Migration(migrations.Migration):
+    def migrate_periodic_tasks(apps, schema_editor):
+        ScheduledMessage = apps.get_model('events', 'ScheduledMessage')
+        EventAction = apps.get_model('events', 'EventAction')
+        Experiment = apps.get_model('experiments', 'Experiment')
+        Participant = apps.get_model('experiments', 'Participant')
+        PeriodicTask = apps.get_model('django_celery_beat', 'PeriodicTask')
+        current_time = datetime.now()
+
+        for task in PeriodicTask.objects.filter(name__startswith="reminder-", enabled=False):
+            task_kwargs = json.loads(task.kwargs)
+            participant_identifiers = task_kwargs["chat_ids"]
+            message = task_kwargs["message"]
+            experiment = Experiment.objects.filter(public_id=task_kwargs["experiment_public_id"])
+
+            frequency, time_period, repetitions, total_triggers, next_trigger_date = calc_schedule_components(task, current_time)
+
+            event_action_params = {
+                "name": "system-generated-event-action",
+                "prompt_text": message,
+                "frequency": frequency,
+                "time_period": time_period,
+                "repetitions": repetitions,
+                "experiment_id": experiment.id
+            }
+
+            for identifier in participant_identifiers:
+                participant = Participant.objects.get(team=experiment.team, identifier=identifier)
+
+                event_action = EventAction.objects.create(
+                    action_type=EventActionType.SCHEDULETRIGGER,
+                    params=event_action_params
+                )
+                ScheduledMessage.objects.create(
+                    experiment=experiment,
+                    participant=participant,
+                    team=experiment.team,
+                    action=event_action,
+                    last_triggered_at=task.last_run_at,
+                    next_trigger_date=next_trigger_date,
+                    total_triggers=total_triggers
+                )
+
+    dependencies = [
+        ('events', '0013_alter_scheduledmessage_action'),
+    ]
+
+    operations = [
+    migrations.RunPython(migrate_periodic_tasks),
```

Comment:
nit: make it elidable
```suggestion
    migrations.RunPython(migrate_periodic_tasks, elidable=True),
```